x-pack/plugins/observability/public/components/shared/exploratory_view/embeddable/use_actions.ts

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { createExploratoryViewUrl } from '../configurations/utils';
+import { ReportViewType } from '../types';
+import { AllSeries } from '../hooks/use_series_storage';
+import { useKibana } from '../../../../../../../../src/plugins/kibana_react/public';
+import {
+  Action,
+  ActionExecutionContext,
+} from '../../../../../../../../src/plugins/ui_actions/public';
+
+export type ActionTypes = 'explore' | 'save' | 'addToCase';
+
+export function useActions({
+  withActions,
+  attributes,
+  reportType,
+  setIsSaveOpen,
+  setAddToCaseOpen,
+  appId = 'observability',
+}: {
+  withActions?: boolean | ActionTypes[];
+  reportType: ReportViewType;
+  attributes: AllSeries;
+  appId?: 'security' | 'observability';
+  setIsSaveOpen: (val: boolean) => void;
+  setAddToCaseOpen: (val: boolean) => void;
+}) {
+  const [defaultActions, setDefaultActions] = useState(['explore', 'save', 'addToCase']);
+
+  useEffect(() => {
+    if (withActions === false) {
+      setDefaultActions([]);
+    }
+    if (Array.isArray(withActions)) {
+      setDefaultActions(withActions);
+    }
+  }, [withActions]);
+
+  const { http, application } = useKibana().services;
+
+  const href = createExploratoryViewUrl(
+    { reportType, allSeries: attributes },
+    http?.basePath.get(),
+    appId
+  );
+
+  const hrefPath = createExploratoryViewUrl(
```

Comment:
`href` and `hrefPath` seem to be equal to me when running this, is there a reason we need to call `createExploratoryViewUrl` twice? Are there cases where `onlyPath` being `undefined` will cause the first call to yield a divergent output?

Reply:
@justinkambic i have removed the part about undefined description.

Reply:
`hrefPath` is actually a route path.  i think there is some room for improvement, i will add another function instead of modifying the current one.

Reply:
@justinkambic i updated the code to make it bit more clear